netmaster/master/netmaster.go

```diff
@@ -501,17 +501,26 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 		// XXX: do configuration check, to make sure it is allowed
 	}
 
+	subnetIsAllocated := false
 	if nwCfg.SubnetIP == "" {
 		nwCfg.SubnetLen = gCfg.Auto.AllocSubnetLen
 		nwCfg.SubnetIP, err = gCfg.AllocSubnet(rm)
 		if err != nil {
 			return err
 		}
+		subnetIsAllocated = true
 	}
 
 	nwCfg.DefaultGw = network.DefaultGw
-	if nwCfg.DefaultGw == "" {
-		// TBD: allocate per global policy
+	// For auto derived subnets assign gateway ip be the last valid unicast ip the subnet
+	if nwCfg.DefaultGw == "" && subnetIsAllocated {
```

Comment:
> Sure, this change, though small, adds a feature imo. I am not sure if we should add just a part of feature, but up to you.

I can attempt to define a new attribute that allows for tagging an endpoint as 'default'. Doing so would also mean that if there is only one interface this is not needed because that is the default interface.
Also note that this is only a problem when not integrating with dockcontivnet, because libnetwork takes care of adding the only interface as a default interface.

> @jainvipin I actually meant adding a few system-tests. I was thinking we should be able to test this feature may be by a ping to default-gateway or routing between networks?

We have routed-mode and bridge mode: 
- This change is for bridged mode, for example in ACI, where we can punt all traffic up and need default gateway there. For 'bridged mode' to test this I may have to simulate an external router that is outside. We need to cover those perhaps as part of those end-to-end system tests.
- In routed mode, @shaleman can confirm that we are already adding the default gateway appropriately, otherwise 'any to any' connectivity wouldn't work. I need to check if he is adding a default route and if so, can that be combined with this change.